nobl9/historical_data_retrieval_schema.go

```diff
@@ -67,9 +79,14 @@ func marshalHistoricalDataRetrieval(d resourceInterface) *v1alpha.HistoricalData
 	historicalDataRetrieval := hData.([]interface{})[0].(map[string]interface{})
 	defaultDuration := historicalDataRetrieval["default_duration"].([]interface{})[0].(map[string]interface{})
 	maxDuration := historicalDataRetrieval["max_duration"].([]interface{})[0].(map[string]interface{})
+	triggeredBySloCreation :=
```

Comment:
I'm getting an error if I don't pass this fields. I did some research and it seems to me that since both of these parameters are optional, assigning them like this will be the cause. A check must be added to see if this field is present in the schema - similar to what is done for SLO: https://github.com/nobl9/terraform-provider-nobl9/blob/main/nobl9/resource_slo.go#L588